app/controllers/mass_rollouts_controller.rb

```diff
@@ -148,6 +145,16 @@ def create_stage_with_group(template_stage)
     stage
   end
 
+  def deploy_stage_reference(template_stages, stage)
+    template_stage = template_stages.detect { |ts| ts.project_id == stage.project.id }
```

Comment:
can turn this into a nice 1-line `return unless template_stage =`
or even better:
`template_stages.detect { |ts| ts.project_id == stage.project.id }&.last_successful_deploy&.reference`